# Genome assembly and variant analysis of *Salmonella enterica* from nanopore long reads

## Introduction
The development of genome sequencing and assembly technology has enabled foundational discoveries across many fields of biology. One of the most remarkable breakthroughs was the assembly of the human genome, which has led to a better understanding of genetic variation and disease mechanisms (1). In addition to humans, sequencing and assembling bacterial genomes has deepened the understanding of virulence factors and pathogen evolution, particularly for clinically relevant species such as Escherichia coli (2) and Salmonella enterica (3).

De novo assembly refers to the recovery of an organism's genome without mapping sequenced DNA fragments to a reference genome. Unlike reference mapping, de novo assembly avoids reference bias, enables the discovery of novel sequences, and facilitates the identification of structural variation (4). However, the process can present notable challenges that hinder the construction of complete, accurate genomes. The most notable challenge with de novo assembly has been balancing the length of sequenced reads with the accuracy of them.

Until recently, short read sequencing technology such as Illumina NGS was the gold standard for producing highly accurate reads used for genome assembly, while long read sequencing technology from Pacific Biosciences (PacBio) and Oxford Nanopore Technologies (ONT) made scaffolding easier but still required a hybrid approach for accuracy in assembly (5). With the development of more accurate, and affordable long read sequencing technology, it is becoming less necessary to use hybrid approaches for de novo assembly, particularly for bacterial species (6).

The current analysis will focus on the methods appropriate for assembly of the *Salmonella enterica*, (*S. enterica*) genome using raw sequence data generated by ONT R10.4. *S. enterica* is a clinically significant foodborne pathogen that causes millions of cases of enterovirus yearly, sometimes leading to severe illness or death (7). Additionally, its single circular chromosome, repeat complexity, and well-characterized reference strains make it useful as a model organism in studies evaluating genome assembly tools (5,8).

ONT long-read sequencing technology works by guiding a DNA or RNA molecule through a protein nanopore fixed to a flow cell membrane. As the molecule passes through the channel, adjacent nucleotides influence the electrical current, producing a signal that is translated into a nucleotide sequence. In contrast to short-read sequencing technologies, the DNA molecule does not need to be fragmented or amplified, enabling the formation of long, contiguous sequences (9).