# Genome assembly and variant analysis of *Salmonella enterica* from nanopore long reads

## Introduction
The development of genome sequencing and assembly technology has enabled foundational discoveries across many fields of biology. One of the most remarkable breakthroughs was the assembly of the human genome, which has led to a better understanding of genetic variation and disease mechanisms (1). In addition to humans, sequencing and assembling bacterial genomes has deepened the understanding of virulence factors and pathogen evolution, particularly for clinically relevant species such as Escherichia coli (2) and Salmonella enterica (3).

De novo assembly refers to the recovery of an organism's genome without mapping sequenced DNA fragments to a reference genome. Unlike reference mapping, de novo assembly avoids reference bias, enables the discovery of novel sequences, and facilitates the identification of structural variation (4). However, the process can present notable challenges that hinder the construction of complete, accurate genomes. A notable challenge in de novo assembly is balancing read length and accuracy. Short read sequencing produces highly accurate reads, but building long contiguous sequences is challenging because repetitive elements are longer than the reads. Even with high sequencing coverage, short reads may not span these repeats, limiting the completeness of the assembly (5). Contrastingly, long read sequencing technologies from Pacific Biosciences and Oxford Nanopore Technologies (ONT) produce reads that span repeats in the genome that recover longer contiguous sequences, but have until recently been expensive and unreliable due to random and systematic errors such as inaccurate homopolymer detection (6). 

ONT long-read sequencing works by guiding a DNA or RNA molecule through a protein nanopore fixed to a flow cell membrane. As the molecule passes through the channel, multiple adjacent nucleotides influence the electrical current, producing a signal that is translated into a nucleotide sequence. In contrast to short-read sequencing technologies, the DNA molecule does not need to be fragmented or amplified, enabling the formation of long, contiguous sequences (7). With their latest R10 chemistry, the nanopore channel is longer, allowing for dual sensors that better inform the electrical signals used for basecalling (6).

The current analysis will focus on the methods appropriate for assembly of the Salmonella enterica genome using raw sequence data generated by ONT R10.4. S. enterica is a clinically significant foodborne pathogen that causes millions of cases of enterovirus yearly, sometimes leading to severe illness or death (8). Additionally, its single circular chromosome, repeat complexity, and well-characterized reference strains make it useful as a model organism in studies evaluating genome assembly tools (9, 10).