# Genome assembly and variant analysis of *Salmonella enterica* from nanopore long reads

## Introduction
Introduction
The development of whole genome sequencing and assembly technology has enabled foundational discoveries for understanding of virulence factors and pathogen evolution, particularly for clinically relevant species (1). *De novo* genome assembly refers to the recovery of an organism's genome without mapping sequenced DNA fragments to a reference genome. Unlike reference mapping, *de novo* assembly avoids reference bias, enables the discovery of novel sequences, and facilitates the identification of structural variation (2). However, the process can present notable challenges that hinder the construction of complete, accurate genomes. A notable challenge in *de novo* assembly is balancing read length and accuracy. Short-read sequencing produces highly accurate reads, but building long contiguous sequences is challenging because repetitive elements are longer than the reads. Even with high sequencing coverage, short reads may not span these repeats, limiting the completeness of the assembly (3). Contrastingly, long-read sequencing technologies from Pacific Biosciences and Oxford Nanopore Technologies (ONT) produce reads that span repeats in the genome that recover longer contiguous sequences, but have until recently been expensive and unreliable due to random and systematic errors such as inaccurate homopolymer detection (4). 

ONT long-read sequencing works by guiding a DNA or RNA molecule through a protein nanopore fixed to a flow cell membrane. As the molecule passes through the channel, multiple adjacent nucleotides influence the electrical current, producing a signal that is translated into a sequence. In contrast to short-read sequencing, the DNA molecule does not need to be fragmented or amplified, enabling the formation of long, contiguous sequences (5). With their latest R10 chemistry, the nanopore channel is longer, allowing for dual sensors that better inform the electrical signals used for basecalling and produce more accurate results (4).

This analysis focuses on assembling the *Salmonella enterica* genome and comparing its variants to a reference genome using raw sequencing data generated by ONT R10.4. *S. enterica* is a clinically significant foodborne pathogen that causes millions of cases of enterovirus yearly, sometimes leading to severe illness or death (6). Additionally, its single circular chromosome, repeat complexity, and well-characterized reference strains make it useful as a model organism in studies evaluating genome assembly tools (7, 8).

There are many assembly tools designed to recover prokaryote genomes using ONT long-read technology. Among the most common are Canu, NECAT, and Flye. When deciding on a tool to use for prokaryotic genome assembly, it is important to consider accuracy, contiguity, consistency, circularisation, computational efficiency, and ease of use. These tools all produce consistent results amongst different datasets, but differ in the remaining criteria (9, 10). Canu has extensive error correction mechanisms and produces highly contiguous assemblies, but is resource intensive and can produce structural errors if coverage is not sufficient (10). NECAT consistently produces the most contiguous assemblies that often achieve circularity relatively quickly (9, 10), but is cumbersome to use and can produce larger scale errors than the other tools (10). Flye is often referred to as a balanced assembler, as it has high contiguity, and is relatively accurate, but can be memory intensive (9, 10). Furthermore, accuracy and circularisation show notable improvements when polishing tools are subsequently used (9, 10). Overall, the best assembly tool depends on the intended application and the context of the pipeline as a whole.

This analysis will assemble the *S. enterica* genome from raw ONT reads obtained from NCBI using Flye in order to balance computational efficiency, accuracy and contiguity. Reads will be filtered using SeqKit, assembled with Flye, polished with Medaka and assessed for quality using QUAST. The proposed workflow is largely based on the approach of Bogaerts and colleagues (11), who conducted phylogenetic analysis and investigated cluster detection of S. enterica and Neisseria meningitidis. For variant calling, reads will be aligned to the S. enterica reference genome using Minimap2 for alignment, followed by file processing with SAMtools. Variants will be called with Clair3 and results will be visualized in IGV. The pipeline for variant calling is informed by the benchmarking analysis by Hall and colleagues (12) that compared the performance of variant calling tools for ONT data.